File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.java
class Solution {
    public int maxProfit(int[] nums) {
        // profit = selling - buying
        int max = 0, bp = nums[0]; 

        for(int num : nums){
            if(num < bp) bp = num; 
            else{
                int currProfit = num - bp; 
                max = Math.max(currProfit, max); 
            }
        }

        return max; 
    }
}

File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-0.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        p  = 1
        i = 1
        while i < len(nums): 
            if nums[i] != nums[i - 1]: 
                nums[p] = nums[i]
                p += 1
            i += 1
        
        return p
        